### PR TITLE
Fix/cachedCostMatrices

### PR DESCRIPTION
**Pull request by chobobobo**
Cleaning up code related to cost matrix caching and trying to find out why occasionally creeps seem to be ignoring the cost matrix entirely.

### CODE REVIEW

File: room.js [hunk 1]
```diff
@@ -1211,14 +1211,16 @@ mod.getCachedStructureMatrix = function(roomName) {
...
-            return {costMatrix: cache.costMatrix, valid: cacheValid(roomName)};
-        } else {
+            return cache.costMatrix;
+        } else if (cache.serializedMatrix) {
             const costMatrix = PathFinder.CostMatrix.deserialize(cache.serializedMatrix);
             cache.costMatrix = costMatrix;
-            return {costMatrix, valid: cacheValid(roomName)};
+            return costMatrix;
+        } else {
+            Util.logError('Room.getCachedStructureMatrix', `Cached costmatrix for ${roomName} is invalid ${cache}`);
```
Comment (Spedwards) [inline room.js]: Should we delete the cached costmatrix here, or is it handled somewhere else?

Comment (chobobobo) [inline room.js]: Yeah, when this returns undefined a new costMatrix will be generated.

**Comment by chobobobo** [comment]
@Spedwards I've caught the bug now, and discovered a second with how stale was (not) working, could you take a second look at this.

**Comment by chobobobo** [comment]
This branch also now has a significant performance boost bringing it in line with the original efficiency

**Comment by chobobobo** [comment]
this resolves https://github.com/ScreepsOCS/screeps.behaviour-action-pattern/issues/1307

**Pull request merged**